modules/mariadb/mariadb.go

```diff
@@ -0,0 +1,161 @@
+package mariadb
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+const (
+	rootUser            = "root"
+	defaultUser         = "test"
+	defaultPassword     = "test"
+	defaultDatabaseName = "test"
+)
+
+// defaultImage {
+const defaultImage = "mariadb:11.0.3"
+
+// }
+
+// MariaDBContainer represents the MariaDB container type used in the module
+type MariaDBContainer struct {
+	testcontainers.Container
+	username string
+	password string
+	database string
+}
+
+// WithDefaultCredentials applies the default credentials to the container request.
+// It will look up for MARIADB environment variables.
+func WithDefaultCredentials() testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) {
+		username := req.Env["MARIADB_USER"]
```

Comment:
@eddumelendez would you see it convenient to check for MYSQL_ variables here? I could imagine evaluating the vars in this order:

- check for MYSQL
- check for MARIADB (more precedence)

Reply:
interesting! However, I would rather use `withUsername`, `withPassword` and `withDatabase` rather than using the env vars directly. I don't think this is an issue because this is a new module, it would be if it is an existing one and just new utility fn were added but that's not the case. Those moving from GenericContainer to the module will also take that into account, I think. Historically, we haven't received issues about it in tc java. Of course, there is an assumption about using the module but if we suggest to use the module because of the pre configuration it brings then it is ok.

If we are going to plan to support `MARIADB_` env vars because `MYSQL_` can be deprecated then I would respect the default image's precedence.